sunkit_image/utils/NoiseLevel.py

```diff
@@ -0,0 +1,201 @@
+import numpy as np
+from scipy.stats import gamma
+from scipy.ndimage import correlate
+from skimage.util import view_as_windows
+
+
+class NoiseLevelEstimation:
+    def __init__(self, img, patchsize=7, decim=0, conf=1-1E-6, itr=3):
+        """
+        NoiseLevel estimates noise level of input single noisy image.
+
+        Input parameters:
+            img: input single numpy image array
+            patchsize (optional): patch size (default: 7)
+            decim (optional): decimation factor. If you put large number, the calculation will be accelerated. (default: 0)
+            conf (optional): confidence interval to determin the threshold for the weak texture. In this algorithm, this value is usually set the value very close to one. (default: 0.99)
+            itr (optional): number of iteration. (default: 3)
+
+        Calculated parameters:
+            nlevel: estimated noise levels.
+            th: threshold to extract weak texture patches at the last iteration.
+            num: number of extracted weak texture patches at the last iteration.
+            mask: weak-texture mask. 0 and 1 represent non-weak-texture and weak-texture regions, respectively
+
+        Example:
+            estimate = NoiseLevelEstimation(noisy_image_array, patchsize=11, itr=10)
+
+        Python Version: 20180718
+        Python Author: M. Kirk
+
+        Translated from Noise Level Estimation Matlab code: noiselevel.m
+
+        noiselevel.m Copyright (C) 2012-2015 Masayuki Tanaka
```

Comment:
What licence is the original version under?

Reply:
From how I read it, I think we can include it as long as we add in the original license file.

Reply:
looks like BSD 2-clause to me, I agree that we can use it.